### PR TITLE
Patch Pycurl, add url encoding

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
When an image URL consists of non-ASCII characters, a 500 error is raised. We replaced the `requests` package with `pycurl` in the 2.8 release. `pycurl` does not encode the URL and raises an error. This is a regression and we need to fix it.

* **What is the new behavior (if this is a feature change)?**
The URL will be encoded the same as how the `requests` package does. No 500 error will be returned.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)

* **Related Python client changes** (link commit/PR here)

* **Related documentation changes** (link commit/PR here)

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/s2_inference/clip_utils.py [hunk 1]
```diff
@@ -139,10 +139,15 @@ def download_image_from_url(image_path: str, image_download_headers: dict, timeo
     if not isinstance(timeout_ms, int):
         raise InternalError(f"timeout must be an integer but received {timeout_ms} of type {type(timeout_ms)}")
 
+    try:
+        encoded_url = requote_uri(image_path)
```
Comment (papa99do) [inline src/marqo/s2_inference/clip_utils.py]: Is it possible that image_path is already encoded? How do we tell?

Comment (wanliAlex) [reply src/marqo/s2_inference/clip_utils.py]: This function passes the given URL through an unquote/quote cycle to ensure that it is fully and consistently quoted. There are unit tests on these cases (URL with encoded characters, URL with mixed encoded and unencoded characters)

File: src/marqo/s2_inference/clip_utils.py [hunk 2]
```diff
@@ -160,6 +165,26 @@ def download_image_from_url(image_path: str, image_download_headers: dict, timeo
     return buffer
 
 
+def encode_url(url: str) -> str:
```
Comment (farshidz) [inline src/marqo/s2_inference/clip_utils.py:175]: This isn't used except in tests?

Comment (wanliAlex) [reply src/marqo/s2_inference/clip_utils.py:175]: solved

**Review by farshidz — APPROVED** [review]

**Pull request merged**